File: python/hashsetOrHashmap/exercise_242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashmap = {}
        if(len(s) != len(t)):
            return False
        for letter in s:
            if letter in hashmap:
                hashmap[letter]+=1
            else:
                hashmap[letter] = 1
        for letter in t:
            if letter in hashmap:
                hashmap[letter]-=1
                if hashmap[letter]<0:
                    return False
            else:
                return False
        return True

File: java/hashsetOrHashmap/exercise_242.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()){
            return false;
        }
        HashMap<Character, Integer> CountS = new HashMap<>();
        HashMap<Character, Integer> CountT = new HashMap<>();
	    for (char c : s.toCharArray()) {
	        CountS.put(c, CountS.getOrDefault(c, 0) + 1);
	    }
	 
	    for (char c : t.toCharArray()) {
	        CountT.put(c, CountT.getOrDefault(c, 0) + 1);
	    }
        return CountT.equals(CountS);
    }
}